### PR TITLE
Be more like cloudformation

### PR DESCRIPTION
**Pull request by tommyvn**

### CODE REVIEW

File: apply [hunk 1]
```diff
@@ -4,17 +4,43 @@ source common.sh
...
+cli_input_json=$(echo -E "${K8S}" | jq -c '.spec | {AttributeDefinitions, ProvisionedThroughput, StreamSpecification} | with_entries(select(.value))')
+
+# ttl_input_json=$(echo -E "${K8S}" | jq -c '.spec | {TimeToLiveSpecification}')
+
+# shellcheck disable=SC2016
+if [[ -z "${K8S_STATUS_TABLEARN}" ]] ; then
+    echo "no arn recorded, creating/retrieving table" >&2
+
+    # shellcheck disable=SC2016
+    args=$(echo -E "$K8S" | jq -r '.spec as $spec | "" |
```
Comment (cherwin) [inline apply]: Hey what is this doing?

File: apply [hunk 2]
```diff
@@ -4,17 +4,43 @@ source common.sh
...
+        if $spec.GlobalSecondaryIndexes then "\(.) --global-secondary-indexes \($spec.GlobalSecondaryIndexes)" else . end |
+        if $spec.LocalSecondaryIndexes then "\(.) --local-secondary-indexes \($spec.LocalSecondaryIndexes)" else . end')
+
+    # shellcheck disable=SC2086
+    table=$(aws dynamodb create-table --table-name "${table_name}" \
+	                               ${args} \
+	                               --cli-input-json "${cli_input_json}" | \
+				       jq '.TableDescription' || aws dynamodb describe-table --table-name "${table_name}" | jq '.Table')
+
+elif echo -E "${K8S}" | jq -e '. as $k8s | .status | {AttributeDefinitions, ProvisionedThroughput, StreamSpecification} | contains($k8s.spec | {AttributeDefinitions, ProvisionedThroughput, StreamSpecification}) | not' >/dev/null || [[ "${K8S_STATUS_TABLESTATUS}" != "ACTIVE" ]] ; then
```
Comment (cherwin) [inline apply]: Commenting the jq queries would be helpful

**Comment by cherwin** [comment]
LGTM 👍

File: apply (hunk 1 above)
Comment (tommyvn) [reply apply]: I've added comments in the commit I just pushed to this PR but to answer the low level detail of what _this_ does, it stores the `.spec` in a variable called `$spec`, sets the input for the next filter to be an empty string, and then proceeds to conditionally add cli args to that empty string to build up the args for the aws cli command.

**Pull request merged**